src/mach/mod.rs

```diff
@@ -24,21 +24,43 @@ pub fn peek(bytes: &[u8], offset: usize) -> error::Result<u32> {
     Ok(bytes.pread_with::<u32>(offset, scroll::BE)?)
 }
 
-#[derive(Debug)]
+#[derive(Copy,Clone,Eq,PartialEq,Hash)]
+pub enum Entrypoint {
+    None,
+    /// The entrypoint was from an `LC_MAIN` load command (added in 10.8) and specifies an offset in
+    /// the `__TEXT` segment.
+    Offset(u64),
```

Comment:
Heh you keep finding bugs :P

I realized I've never loaded an executable (and for dylibs its always 0)

So I'm wondering if we should relocate the offset based on the vmaddress in the text.

I think the formula is:

`(text_vmaddr - text_offset) + entry_offset`

Actually I just realized after typing this the whole point of your PR was you wanted to distinguish the entrypoint case.

Is this actually super important; e.g., could we just relocate it to be a vm, so that it matches more closely both the PE and ELF entry field?